my-first-project/src/app/hero.service.ts

```diff
@@ -1,25 +1,120 @@
 import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
 
 import { Observable, of } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
 
 import { Hero } from './hero';
-import { HEROES } from './mock-heroes';
 import { MessageService } from './message.service';
 
+const httpOptions = {
+  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+};
+
 @Injectable({ providedIn: 'root' })
 export class HeroService {
 
-  constructor(private messageService: MessageService) { }
+  private heroesUrl = 'api/heroes';  // URL to web api
+
+  constructor(
+    private messageService: MessageService,
+    private http: HttpClient) { }
+
+  /** GET heroes from the server */
```

Comment:
It's just a project preference, but we don't put inline comments inside the .ts or .css or .html files.

For any documentation purposes we add a README.md file